experimental/dds/tree2/src/test/utils.ts

```diff
@@ -1010,3 +1030,53 @@ export function createTestUndoRedoStacks(
 
 	return { undoStack, redoStack, unsubscribe };
 }
+
+/**
+ * Mock IdCompressor for testing that returns an incrementing ID.
+ * Should not be used for tests that have multiple clients as doing so will generate
+ * duplicate IDs that collide.
+ */
+export class MockIdCompressor implements IIdCompressor, IIdCompressorCore {
```

Comment:
This class is probably unnecessary. Cases that use this could probably just call `createNewCompressor`. It was created when fixing tests that can't use the 'global' test compressor (see below) but also don't connect with multiple clients.